HCCGo/app/js/jobStatusService.js

```diff
@@ -1,8 +1,154 @@
 jobStatusService = angular.module('jobStatusService', []);
 
-jobStatusService.service('jobStatusService', function() {
-
+jobStatusService.service('jobStatusService',['$log','$q','notifierService', function($log, $q, notifierService) {
+	var async = require('async');
+	var oldData = null;
+	var lastRequestedTime = 0;
+	// Refresh every 15 seconds
 	return {
-		
+		refreshDatabase: function(db, clusterInterface, clusterId, force=false) {
+			var toReturn = $q.defer();
+			if (Date.now() - lastRequestedTime >= 15000 || force){
```

Comment:
Can you flip this if around, so the `toReturn.resolve(oldData)` is at top here, and all this async stuff is in the `else`